### PR TITLE
33 documentation on api 

### PR DESCRIPTION
**Pull request by owitiA**
#### Pre-Submission Checklist

- [x] Your pull request targets the `stage` branch of Kudobuzz.
- [x] Branch name is descriptive. 
- [x] All new and existing tests pass the command `npm test`. Use `git commit --amend` to amend any fixes.

#### Checklist:

- [ ] All test are passing. 
- [ ] Tested changes locally.
- [ ] Api Documentation follows api standards.
- [ ] Improved documentation or readme where necessary. 
- [ ] Your commits messages format follows our commit message guideline.

#### What does this pr do?
Implement the API for To do's and documents the API to open api 3.0.0

### CODE REVIEW

File: api/README.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
+# To Do API
```
Comment (obengwilliam) [inline api/README.md]: @owitiA  are u done with this ?

Comment (owitiA) [reply api/README.md]: the documentation is the first part and i am working on the API now

File: api/api.yaml [hunk 2]
```diff
@@ -0,0 +1,61 @@
+swagger: "2.0"
```
Comment (obengwilliam) [inline api/api.yaml]: https://github.com/kudobuzz/api-standards
Do take a look at the api standard

Comment (owitiA) [reply api/api.yaml]: yes i wrote it in swagger first to better understand documenting api's will work on making it open api 3.0 today

**Comment by owitiA** [comment]
update on docs ,followed the api standards docs, kindly check if its the right way of executing the docs

File: api/api.yaml [hunk 3]
```diff
@@ -1,61 +1,90 @@
-swagger: "2.0"
+openapi: 3.0.0
```
Comment (obengwilliam) [inline api/api.yaml]: Screen Shot 2019-03-27 at 1 35 44 PM
There is an error

File: api/api.yaml [hunk 4]
```diff
@@ -1,61 +1,90 @@
...
 info:
-  title: A Todo list application
-  version: 1.0.0
-
+  version: 'v1'
+  title: 'A simple API introduction '
```
Comment (obengwilliam) [inline api/api.yaml]: Kindly update title

File: api/api.yaml [hunk 5]
```diff
@@ -1,61 +1,90 @@
...
 schemes:
-- http
-- https
+  http
```
Comment (obengwilliam) [inline api/api.yaml]: Again follow api standards, Look at how responses are suppose to be formatted

File: api/api.yaml (hunk 3 above)
Comment (owitiA) [reply api/api.yaml]: yes i was just about to tell you that error is persisiting no matter what i change looked line by line even had kevin look at it for me

File: api/api.yaml (hunk 4 above)
Comment (owitiA) [reply api/api.yaml]: ok

**Comment by obengwilliam** [comment]
There is still quite a number of things you are missing

Model for a todo
title, string, required
note, string, required
due_date, string, required
created_at, string, required
updated_at, string, required


So up there is an ideal todo

## Api 

So GET /todos  should return a list of todos 
like return  {
  data: [
   {
  // todo goes here
}
],
metadata: {
  count:  4
}
}


POST /todos
request schema
 {
 title // must string and required
 due date/ must be string and required and in the future
 note // string and must be more than 30 characters
}

response schema
 {
   data: {
     title
due_date
note  
// this should be the data that was just created
}
}

DELETE /todos/:todoid

todoid // is the id of the todo
response

{
   data: {
  should return deleted todo
}
}

PUT /todo/: todoid

request schema  can be optional values 
which can be title, note and due date , one of them is required


and response will be the todoid that was updated
as {data: {todo goes here}}

**Comment by obengwilliam** [comment]
@owitiA  look at the example

File: api/README.md [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+
+This is a simple api for  interacting with the boilerplate
```
Comment (obengwilliam) [inline api/README.md]: rather add a link to the actual api, instead of documenting it here

File: api/api.yaml (hunk 3 above)
Comment (obengwilliam) [reply api/api.yaml]: ?

File: api/README.md (hunk 6 above)
Comment (owitiA) [reply api/README.md]: done

File: api/README.md (hunk 1 above)
Comment (owitiA) [reply api/README.md]: done

File: api/api.yaml (hunk 3 above)
Comment (owitiA) [reply api/api.yaml]: i don't understand

File: api/api.yaml (hunk 5 above)
Comment (owitiA) [reply api/api.yaml]: done

File: api/api.yaml (hunk 3 above)
Comment (obengwilliam) [reply api/api.yaml]: you dont understand what ?

i am just asking if that has been fixed

File: api/api.yaml [hunk 7]
```diff
@@ -0,0 +1,218 @@
...
+          description: Array of Todo model instances
+          examples:
+            application/json:
+              schema:
+                $ref: '#/definitions/todosObjectinDB '
+        examples:
+            application/json:
+            {
+              data: {
+                    "todos": [
```
Comment (obengwilliam) [inline api/api.yaml]: Take a look athe api standards, it should not be data: {
todods: []}  ut just  {data: []}

File: api/health.yaml [hunk 8]
```diff
@@ -0,0 +1,37 @@
+openapi: 3.0.0
```
Comment (obengwilliam) [inline api/health.yaml]: health route is still a route it should be part of api.yaml

Comment (owitiA) [reply api/health.yaml]: done

File: api/api.yaml (hunk 3 above)
Comment (obengwilliam) [reply api/api.yaml]: ?

**Pull request merged**